compiler/cli/src/org/jetbrains/kotlin/cli/common/Usage.java

```diff
@@ -53,6 +53,8 @@
             appendln(sb, "Advanced options are non-standard and may be changed or removed without any notice.");
         }
         else {
+            if(tool.executableScriptFileName().equals("kotlinc-jvm"))
```

Comment:
Why is this check needed? Isn't `-J` also supported for JS and Native compilers?